lib/travis/build/script/elm.rb

```diff
@@ -0,0 +1,125 @@
+module Travis
+  module Build
+    class Script
+      class Elm < NodeJs
+        # Default NodeJS version to install
+        DEFAULT_VERSION = '6.10.0'
+
+        DEFAULTS = {
+          elm_version: 'latest',
+          elm_test_version: 'latest'
+        }
+
+        ELM_TEST_REQUIRED_NODE_VERSION = '4.0.0'
+
+        def export
+          super
+          sh.export 'TRAVIS_ELM_VERSION', elm_version, echo: false
+          sh.export 'TRAVIS_ELM_TEST_VERSION', elm_test_version, echo: false
+        end
+
+        def configure
+          super
+
+          sh.if '! -d sysconfcpus/bin' do
+            install_sysconfcpus
+          end
+        end
+
+        def announce
+          super
+          sh.cmd 'elm --version', echo: true
+          sh.cmd 'elm-test --version', echo: true
+        end
+
+        def setup
+          super
+
+          install_elm
+          install_elm_test
+        end
+
+        def script
+          sh.cmd 'elm-format --validate .'
+          sh.cmd 'elm-test'
+        end
+
+        def setup_cache
+          if data.cache?(:elm)
+            sh.fold 'cache.elm' do
+              directory_cache.add '$HOME/.cache/elm-stuff', 'elm-stuff'
+
+              # some put their tests in ./test and others use ./tests
+              directory_cache.add '$HOME/.cache/elm-test-stuff', 'test/elm-stuff'
+              directory_cache.add '$HOME/.cache/elm-tests-stuff', 'tests/elm-stuff'
+
+              # we build sysconfcpus from source, so cache the result
+              directory_cache.add '$HOME/.cache/sysconfcpus-cache', 'sysconfcpus'
+            end
+          end
+        end
+
+        def cache_slug
+          super << '-elm-' << elm_version
+        end
+
+        private
+
+          def elm_version
+            config[:elm_version].to_s
+          end
+
+          def elm_test_version
+            config[:elm_test_version].to_s
+          end
+
+          def install_elm
+            npm_install '-g elm-use@">=1.1.0 <2.0.0"'
+            sh.cmd "elm-use #{elm_version}", retry: true
```

Comment:
`elm-use` 1.1.1 returns `0`; I just upgraded the version range to that. Should be fixed now!